app/src/DataFetcher.tsx

```diff
@@ -0,0 +1,49 @@
+// Note: temporary class for prototyping. Will be replaced with real API calls.
+
+class DataFetcher {
+  async loadData_(url: string) {
+    const r = await fetch(url);
+    return await r.json();
+  }
+
+  getUrl_(datasetId: string) {
```

Comment:
@aaronsn you probably know this already but you can also list class members as `private`, rather than using the `_` approach, e.g. 

```ts
class DataFetcher {
  private async loadData(url: string) {
    // . . .
  }
}
```

that way Typescript will yell at you if you try to use the private members in the class instance.

Reply:
I.... forgot about this haha. Haven't used typescript all that much. Updated to private methods